modules/runtime/nuxeo-runtime-test/src/test/java/org/nuxeo/runtime/RuntimeInitializationTest.java

```diff
@@ -47,18 +50,111 @@ public void testContributions() throws Exception {
+            assertEquals(0, errors.size());
+        }
+    }
+
     @Test
     @Deploy("org.nuxeo.runtime.test.tests:MyComp1.xml")
     @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
-    @Ignore("Deactivated for now since duplicate contributions are still allowed.")
+    @Deploy("org.nuxeo.runtime.test.tests:CopyOfMyComp2.xml")
     public void testContributionsWithDuplicateComponent() throws Exception {
-        boolean success = false;
-        try {
-            hotDeployer.deploy("org.nuxeo.runtime.test.tests:CopyOfMyComp2.xml");
-            success = true;
-        } catch (AssertionError e) {
-            // OK.
-        }
-        assertFalse("An exception should have been raised.", success);
+        checkDupe(true);
+    }
+
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp1.xml")
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
+    public void testContributionsWithDuplicateComponentSameFile() throws Exception {
+        // dupe annotation on tests not detected (?)
+        checkDupe(false);
     }
+
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp1.xml")
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
+    public void testContributionsWithDuplicateComponentSameFileHotReload() throws Exception {
+        hotDeployer.deploy("org.nuxeo.runtime.test.tests:MyComp2.xml");
+        checkDupe(true);
+    }
+
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp1.xml")
+    @Deploy("org.nuxeo.runtime.test.tests:MyComp2.xml")
+    public void testContributionsWithDuplicateComponentHotReload() throws Exception {
+        hotDeployer.deploy("org.nuxeo.runtime.test.tests:CopyOfMyComp2.xml");
+        checkDupe(true);
+    }
+
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:invalid-component-activate-message.xml")
+    public void testInvalidComponentActivateMessage() throws Exception {
+        assertEquals(List.of("Error message on activate"),
+                Framework.getRuntime().getMessageHandler().getMessages(Level.ERROR));
+        assertEquals(List.of("Warn message on activate"),
+                Framework.getRuntime().getMessageHandler().getMessages(Level.WARNING));
+    }
+
+    @Test
+    @Deploy("org.nuxeo.runtime.test.tests:invalid-component.xml")
+    public void testInvalidComponent() throws Exception {
+        assertEquals(List.of(
+                "Bad extension declaration (no target attribute specified) on component 'service:invalid.comp'",
+                "Warning: target extension point 'xp' of 'invalid.comp' is unknown. Check your extension in component service:invalid.comp",
+                "Warning: target extension point 'null' of 'invalid.comp' is unknown. Check your extension in component service:invalid.comp"),
+                Framework.getRuntime().getMessageHandler().getMessages(Level.ERROR));
+        assertEquals(List.of(), Framework.getRuntime().getMessageHandler().getMessages(Level.WARNING));
```

Comment:
```suggestion
        assertTrue(Framework.getRuntime().getMessageHandler().getMessages(Level.WARNING).isEmpty());
```

Same for others `assertEquals(List.of(), ...)`